tqdm/_tqdm.py

```diff
@@ -734,6 +752,10 @@ def __iter__(self):
                                 else smoothing * delta_t / delta_it + \
                                 (1 - smoothing) * avg_time
 
+                        # Acquire locks if parallel bars
```

Comment:
Er, I learnt through trials and errors mostly (and because I myself tried to unify ``__iter__()`` and ``update()``, and I spent a long time...).

Yes copying the method is not the most difficult part, the difficult part is to dynamically rewrite all statements that use instance variables to use local variables instead. I think if we go down this road that this should be done by modifying the AST to preserve efficiency (so that it gets rewritten once at class instanciation). But then that's really a lot of work just to avoid one method duplication...